### DISCUSSION TITLE
Rule action/trigger value

### DISCUSSION THREAD

**Discussion by adamko01** [firefly-iii/firefly-iii#9331] [General]
Hi guys,
my question would if there is possibility to create this rule:
If Description value is empty or (empty description)
Action Set Description to Transaction Source account.

This is because in my .csv there is no description set and I also cant map the Source account as description and at the same time partner name.

**Comment by JC5** [comment]
That's possible, yes.

- Rule trigger: Description is... "(no description)"
- Rule action "Set description to" `=source_account_name`

Use the `=` sign and it should work, let me know.

**Reply by adamko01** [reply]
Thank you! Working like a charm.

**Reply by vekvoid** [reply]
>     * Rule action "Set description to" `=source_account_name`
Is this the full list of available placeholders for rule actions?

```
'amount'
'date'
'source_account_id'
'budget_name'
'category_name'
'source_account_name'
'destination_account_id'
'destination_account_name
'description'
'transaction_group_id'
```

[Source](https://github.com/firefly-iii/firefly-iii/blob/af3b40a314f3ebbd86ece06b68bfccfe4e17e493/app/Repositories/Tag/OperationsRepository.php#L99)

**Reply by JC5** [reply]
https://docs.firefly-iii.org/references/firefly-iii/rule-expressions/
